pkg/utils/predicates.go

```diff
@@ -275,3 +275,29 @@ func TolerationsEqual(t1, t2 []v1.Toleration) bool {
 	}
 	return true
 }
+
+// Returns the weight that the pod gives to a node by analyzing the
+// soft node affinity of that pod
+// (nodeAffinity.preferredDuringSchedulingIgnoredDuringExecution)
+func PodNodeAffinityWeight(pod *v1.Pod, node *v1.Node) (int32, error) {
```

Comment:
Can we be more explicit in the function name that this only checks preferred?

Reply:
The function has been renamed to `GetNodeWeightGivenPodPreferredAffinity`